avm/res/event-hub/namespace/tests/e2e/encr/main.test.bicep

```diff
@@ -53,7 +53,7 @@ module testDeployment '../../../main.bicep' = {
   scope: resourceGroup
   name: '${uniqueString(deployment().name, resourceLocation)}-test-${serviceShort}'
   params: {
-    name: '${namePrefix}${serviceShort}001'
+    name: '${namePrefix}${serviceShort}${substring(uniqueString(baseTime), 0, 3)}'
```

Comment:
Hey @segraef could you elaborate why this was changed? Just curious, as it take away the possiblity to test this deployment in an idempontent way.

Reply:
Hi @AlexanderSehr the unique name should solve a conflict when deploying the encrypted eventhub:
  VERBOSE: 11:05:03 - The deployment 'a-r-eh-n-encr-t1-20240612T1106073886Z' failed with error(s). Showing 1 out of 1 error(s).
  Status Message: At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/arm-deployment-operations for usage details. (Code: DeploymentFailed)
   - The resource write operation failed to complete successfully, because it reached terminal provisioning state 'Failed'. (Code:ResourceDeploymentFailure)

Reply:
Ok, but when I take a look at the last failed pipelines I see 3 erros through 3 attempts

## 1. 
```
The deployment 'a-r-eh-n-encr-t1-20240612T1106073886Z' failed with error(s). Showing 1 out of 1 error(s).
  Status Message: At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/arm-deployment-operations for usage details. (Code: DeploymentFailed)
   - The resource write operation failed to complete successfully, because it reached terminal provisioning state 'Failed'. (Code:ResourceDeploymentFailure)
```

## 2.
```
The deployment 'a-r-eh-n-encr-t2-20240612T1106082182Z' failed with error(s). Showing 2 out of 2 error(s).
  Status Message: Namespace name '***ehnenc001' is not available. Reason: NameInUse. Message: Name already in use (Code:NamespaceUnavailable)
  
  Status Message: At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/arm-deployment-operations for usage details. (Code: DeploymentFailed)
   - Namespace name '***ehnenc001' is not available. Reason: NameInUse. Message: Name already in use (Code:NamespaceUnavailable)
```

## 3.
```
The deployment 'a-r-eh-n-encr-t3-20240612T1106487953Z' failed with error(s). Showing 2 out of 2 error(s). Status Message: Namespace name '***ehnenc001' is not available. Reason: NameInUse. Message: Name already in use (Code:NamespaceUnavailable)
  Status Message: At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/arm-deployment-operations for usage details. (Code: DeploymentFailed)  
   - Namespace name '***ehnenc001' is not available. Reason: NameInUse. Message: Name already in use (Code:NamespaceUnavailable)
```

Of which only the first is really relevant as all others could be / are follow-up errors as the Namespace from Attempt 1 is still around (and later the one from Attempt 2. 
I don't think we're actually fixing the error, but just masking it by deploying 3 different Namespaces. 

I would suggest to run the deployment of the `encr` test once without the removal (which will likely fail) and then look into the deployment and find the actual cause of the error. `The resource write operation failed to complete successfully` isn't exactly saying much. 
My guess is that we have similar issues like we have with other modules where the permissions for the key vault key take longer to propagate than the deployment is willing to wait (which is super nasty by the way). If that turns out to be the case here too, then we should fix that instead. Likely by adding some wait logic via an inline deployment script, as annoying as it may be.

Am I making sense 😄 ?

Reply:
Found that the issue on deployment failure was related to the publicNetworkAccess parameter set to "SecuredByPerimeter". Solved in PR #2361 
Setting this PR to draft, we'll use this for the API update

Reply:
Wow, look at this who could've guess this. Great work thanks @elanzel !